src/main/java/org/spongepowered/common/mixin/core/inventory/ContainerMixin.java

```diff
@@ -513,17 +512,6 @@ private ItemStack redirectTransferStackInSlot(final Container thisContainer, fin
                 result = ItemStack.EMPTY; // Return empty to stop shift-crafting
             }
         }
-
-        final Inventory craftInv = ((Inventory) thisContainer).query(QueryOperationTypes.INVENTORY_TYPE.of(CraftingInventory.class));
```

Comment:
Just want some clarity, how is deleting this changing behavior? Wouldn't technically the preview transactions no longer be populated in this case?

Reply:
Lines here was added in https://github.com/SpongePowered/SpongeCommon/commit/a3fac249b400056ca2544372dd0f01a9c28aa280 for https://github.com/SpongePowered/SpongeCommon/issues/2384.
But I find that the event is fired. but the SlotTransaction created at https://github.com/SpongePowered/SpongeCommon/blob/f7bde6a121d20b0f043b0d96ffd5edca711e103a/src/main/java/org/spongepowered/common/mixin/core/inventory/SlotCraftingMixin.java#L183 is not added to the SlotTransctionList. So the custom result will not be handled at https://github.com/SpongePowered/SpongeCommon/blob/f7bde6a121d20b0f043b0d96ffd5edca711e103a/src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java#L1465